protocol/0087-EVMD-eth-rpc-and-evm-data-source.md

```diff
@@ -0,0 +1,129 @@
+- Ensure that when a market data source type is amended from internal, external, ethereum or open (coinbase) to an alternative for both termination and settlement we see that old data source is deactivated (if no other market is using) and we see the new data source created and it supports termination and settlement specific to its data source type (<a name="0087-EVMD-007" href="#0087-EVMD-007">0087-EVMD-007</a>)
+
+
+### External Oracles - Validations
+
+- A market proposal that reference an `EthRpcEvmCompatible` that's not active will fail at enactment stage (<a name="0087-EVMD-008" href="#0087-EVMD-008">0087-EVMD-008</a>).
+- Validate if the smart contract address is valid (<a name="0087-EVMD-009" href="#0087-EVMD-009">0087-EVMD-009</a>)
+- Validate if the data elements of the oracle data source is valid - e.g. call the smart contract and check if the types in the ABI match whats provided in the oracle spec (<a name="0087-EVMD-010" href="#0087-EVMD-010">0087-EVMD-010</a>)
+- Validations for min / max frequency of listening for events / read a smart contract (<a name="0087-EVMD-011" href="#0087-EVMD-011">0087-EVMD-011</a>)
+- When a proposal that uses ethereum oracles, defines incorrect data (contract address, ABI) the system should return an error and the proposal should not pass validation (<a name="0087-EVMD-012" href="#0087-EVMD-012">0087-EVMD-012</a>)
+- Any mismatch between expected fields/field types and received fields/field types should emit an error event (<a name="0087-EVMD-013" href="#0087-EVMD-013">0087-EVMD-013</a>)
+
+
+### Usage
+
+- Two different markets may reference two identical `EthRpcEvmCompatible` contracts and ABIs, even with same contract address *but* on two different Ethereum L2s and they each get the correct values i.e. if we have market 1 with
+
+```yaml
+source: EthereumEvent {
+            source_chain_id: 0x123
+            contract: 0xDEADBEEF
+            ABI: "...JSON..."
+            event: "MyFaveEvent"
+```
+
+and market 2 with
+
+```yaml
+source: EthereumEvent {
+            source_chain_id: 0x789
+            contract: 0xDEADBEEF
+            ABI: "...JSON..."
+            event: "MyFaveEvent"
+```
+
+Then market 1 only sees events of that type from `EthRpcEvmCompatible` 0x123 while market 2 only sees events of that type from `EthRpcEvmCompatible` 0x789 (<a name="0087-EVMD-014" href="#0087-EVMD-014">0087-EVMD-014</a>).
+
+- It should be possible to use only `EthRpcEvmCompatible` data sources in a market proposal, or create any combination with any of the other types of currently existing external or internal data sources (<a name="0087-EVMD-015" href="#0087-EVMD-015">0087-EVMD-015</a>)
```

Comment:
These are all very future's focussed. Could we make sure we have a happy path AC for perpetuals? And maybe also one for doing a `UpdateMarket` from a data-source on normal Ethereum to a data-source from an EVM-compatible chain?

Reply:
Both very good points @wwestgarth let me add them.